### ISSUE TITLE
Add support for text underline/overline/strikeout

### ISSUE THREAD

**Issue by tronical** [slint-ui/slint#3192]
Besides the existing support for `font-italic`, `font-weight`, and `font-size`, we should also support text line decorations, such as underline, overline, strikeout.

The number of properties are becoming a little too much, so part of the acceptance criteria for this issue should also be that we can keep a nice API with what looks like individual properties, but in the implementation attempt to consolidate these into fewer `Property` instances.

**Comment by ogoffart** [comment]
We can have a pass that lowers all these property to a struct `{italic: bool, underline:bool, striked: bool, weight: u16, size: length}`
What properties exactly will he have? 
I think it makes sense to keep some property appart like size

**Comment by tronical** [comment]
Yes, size makes sense to keep apart. Combining italic, underline, etc. might be worth it.

I wonder how to handle a case like this:

```
component Foo inherits Text {
    font-underline: { some expression }
}

component Bar {
    in-out property <bool> font-italic; 
    Foo {
        font-underline <=> font-italic;
    }
}
```

I can see how we could perhaps at compile time combine multiple binding expressions into one big binding expression for a struct, but how could this be done when there's a two-way binding on one of the fields?

**Comment by ogoffart** [comment]
Do we still need this now that we have `StyledText`?

**Comment by chrrs** [comment]
> Do we still need this now that we have `StyledText`?

Hi! I ran into this today, and I think this should be re-opened. I think it makes a lot of sense to still add these to `Text`. For example, I have a text element I want to show with an underline when hovered. Something like:

```slint
component TextLink inherits Text {
    touch_area := TouchArea { }
}

TextLink {
    text: "Foo";
    color: #ffffff;
    font-weight: 700;
}
```

To make this seemingly simple change, I'd need to change `color` to `default-color`, `font-weight` doesn't even exist, and the underline would need to become something like `text: underline ? @markdown("<u>\{value}</u>") : value`, which feels incredibly hacky to me. ~~Not to mention that this might open up to `value` injecting unwanted styles.~~ It looks like it's currently impossible to make underlined, semibold text in Slint.

**Comment by ogoffart** [comment]
> `value` injecting unwanted styles

Off-topic, but this is not the case.   interpolation in `@markdown` are properly escaped
